### PR TITLE
use stable v1 version for workload API

### PR DESCRIPTION
**Pull request by wackxu**
Since workloads API has GA since k8s 1.9 version, we should use stable version instead of beta or alpha.

### CODE REVIEW

File: galley/pkg/crd/validation/webhook.go [hunk 1]
```diff
@@ -196,7 +197,7 @@ func NewWebhook(p WebhookParameters) (*Webhook, error) {
 		wh.ownerRefs = []v1.OwnerReference{
 			*v1.NewControllerRef(
 				galleyDeployment,
-				extensionsv1beta1.SchemeGroupVersion.WithKind("Deployment"),
+				appsv1.SchemeGroupVersion.WithKind("Deployment"),
```
Comment (ayj) [inline galley/pkg/crd/validation/webhook.go]: (referencing line 193) - we should also use the appsv1 clientset interface, i.e. `wh.clientset.AppsV1.Deployments(...)`

File: galley/pkg/crd/validation/webhook.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ import (
 	"istio.io/istio/pilot/pkg/config/kube/crd"
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/log"
+	"k8s.io/kubernetes/staging/src/k8s.io/client-go/scale/scheme/extensionsv1beta1"
```
Comment (ayj) [inline galley/pkg/crd/validation/webhook.go]: This doesn't look right, but I can't find any code that is actually referencing it in this file? Stale import?

**Comment by wackxu** [comment]
@ayj Done, PTAL

**Comment by ayj** [comment]
@costinm, `apps/v1` is only available on k8s >= 1.9. With this change, Istio cannot be installed with the default YAMLs on k8s <= 1.8.

**Comment by wackxu** [comment]
@ayj @costinm But  we use mutating webhook admission controller to automatically inject sidecar. This feature requires Kubernetes 1.9 or later. I think it is acceptable for this update.

**Comment by wackxu** [comment]
There is no doubt that we should use `apps/v1` for k8s workload resources because `extention` group has been deprecated for a long time and will be removed in the future and `apps` group has been GA since k8s 1.9 version and k8s in-tree controller has also used `apps/v1` version.

workloads resources have an incompatible change since k8s 1.9 that the field `Selector` in resources must be explicitly specified and must match the pod template's labels. so we need add `Selector` field for every yaml of workload resources, then it is a big job, so I stat the path that need to change
```
 samples\bookinfo\platform\kube
 samples\httpbin
 samples\rawvm
 tests\e2e\tests\simple\testdata\nginx
 tests\e2e\tests\simple\testdata\v1alpha3
 install\kubernetes\helm\istio\charts\certmanager\templates
 install\kubernetes\helm\istio\charts\galley\templates
 install\kubernetes\helm\istio\charts\prometheus\templates
 tests\helm\templates
 install\kubernetes\helm\istio\charts\tracing\templates
 pilot\pkg\kube\inject\testdata\inject
 security\cmd\node_agent_k8s
 install\kubernetes\helm\istio-remote\charts\security\templates
 install\kubernetes\helm\istio\charts\grafana\templates
 install\kubernetes\helm\istio\charts\mixer\templates
 tests\e2e\tests\pilot\testdata
 install\kubernetes\helm\istio\charts\ingress\templates
 install\kubernetes\helm\istio\charts\kiali\templates
 tests\e2e\tests\dashboard
 pilot\pkg\kube\inject\testdata\webhook
 samples\websockets
 install\kubernetes\helm\istio\charts\servicegraph\templates
 istioctl\cmd\istioctl\testdata\deployment
 mixer\test\prometheus\cmd
 samples\httpbin\sample-client
 samples\kubernetes-blog
 install\kubernetes\addons
 install\kubernetes\helm\istio\charts\gateways\templates
 samples\helloworld
 samples\sleep
 tools
 install\kubernetes\helm\istio\charts\pilot\templates
 install\kubernetes\helm\istio\charts\security\templates
 samples\health-check
 pkg\test\framework\components\policybackend
 install\kubernetes\citadel_extras
 install\kubernetes\helm\istio-remote\charts\sidecarInjectorWebhook\templates
 mixer\test\listbackend\cmd
 install\kubernetes\helm\istio\charts\sidecarInjectorWebhook\templates
 tests\helm
```
so we can do this in several step,
1. update go file to use `apps/v1` version, we can do this work in this PR.
2. I will open an issue to trace the work that replace the `extention` to `apps/v1`, and I have several guys to help do this.
3. update rabc yaml.

@ayj @costinm Any thoughts?

**Comment by ozevren** [comment]
Following up on some old PRs.
Was there a final verdict for this PR, based on @wackxu's explanation above?

**Comment by ayj** [comment]
The PR and @wackxu's plan lgtm.

> 1. update go file to use apps/v1 version, we can do this work in this PR.
> 1. I will open an issue to trace the work that replace the extention to apps/v1, and I have several guys can help do this.
> 1. update rabc yaml.

(2) and (3) may need to be combined in the same PR.

**Comment by ozevren** [comment]
@wackxu can you resolve conflicts/update your PR and follow up with your plan?

Thanks

**Comment by wackxu** [comment]
@ozevren Yes， will do it today

**Comment by ayj** [comment]
Looks like the deployment selector is still missing, e.g.
```bash
error: error validating "/tmp/istio.e2e.242718987/yaml/istio-galley.yaml": error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by ayj** [comment]
Hi @wackxu, were you still planning to work on getting this change merged?

**Pull request closed without merging**